Test

TomCohenAbeille opened the pull request:
feat : test if the csv is valid 
refacto : assing_row_to_variables with the index directly 
feat : documenting functions

TomCohenAbeille committed:
Add files via upload

TomCohenAbeille committed:
Add files via upload

feat  : work with the rows in any order 
todo : test if the csv contain all 3 rows (min_expo,max_expo,step)

TomCohenAbeille committed:
Add files via upload

feat : test if the constraints file is valid

TomCohenAbeille committed:
Add files via upload

TomCohenAbeille committed:
Add files via upload

TomCohenAbeille committed:
Add files via upload

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,188 @@
+)
+
+
+from test.constants import ASSETS, MAX_EXPO, MIN_EXPO, STEP  # noqa: F403, E402
```
Il faut mettre ses fichiers de code source dans le dossier src
Le dossier test est normalement réservé aux codes sources de test avec pytest

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,188 @@
+    References
+    ----------
+        [1] The official pandas website : https://pandas.pydata.org/
+        [2] Multiple pages of the StackOverflow websites : https://stackoverflow.com/
```
Très bien pour la docu
Pas besoin de mettre les parties See Also et References, c'est pour les fonctions dans des gros packages ça

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,188 @@
+        print("la colonne", ASSETS, "est présente")
+    else:
+        raise Exception(
+            "Le fichier des contraintes nest pas valide, assurez-vous que la colonne",
```
orthographe : n'est pas valide -> manque l'apostrophe

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,188 @@
+        print("Le fichier des contraintes est valide")
+    else:
+        raise Exception(
+            "Le fichier des contraintes nest pas valide, assurez-vous que",
```
orthographe : de meme

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,188 @@
+
+
+def compute_all_possible_weights(df: pd.DataFrame) -> list:
+    """Compute all the possible weight for all the assets
```
orthographe : s a la fin de weight

KwanV92 reviewed (changes requested):
Attention à soigner l'orthographe, même le code doit être propre, c'est comme si on écrivait un mail aux grands chefs de Abeille haha

TomCohenAbeille committed:
Add files via upload

feat : the user can rename the output file
fix : issue where the last weight is not compute 
docs : cleaning and correcting
